src/Vm.sol

```diff
@@ -1699,6 +1708,14 @@ interface Vm is VmSafe {
     /// Calldata match takes precedence over `msg.value` in case of ambiguity.
     function mockCall(address callee, uint256 msgValue, bytes calldata data, bytes calldata returnData) external;
 
+    /// Whenever a call is made to `callee` with calldata `data`, this cheatcode instead calls
+    /// `target` with the same calldata. This functionality is similar to a delegate call made to
+    /// `target` contract from `callee`.
+    /// Can be used to substitute a call to a function with another implementation that captures
+    /// the primary logic of the original function but is easier to reason about.
+    /// If calldata is not a strict match then partial match by selector is attempted.
+    function mockFunction(address callee, address target, bytes calldata data) external;
```

Comment:
Curious what the use case is for this? 

> If calldata is not a strict match then partial match by selector is attempted.

What does this mean, why is there calldata matching here? I thought it would just be `address(target).call(data)`

Reply:
one could add different mocks for a function called with different params (e.g. mock `Counter.count(1)` with `Counter1.count()` while `Counter.count(2)` mocked with `Counter2.count()`) or a catch all mock for `Counter.count`